taskpools/primitives/barriers_posix.nim

```diff
@@ -18,8 +18,12 @@ when defined(osx):
   export PthreadAttr, PthreadBarrier, Errno, PTHREAD_BARRIER_SERIAL_THREAD
 else:
   type
-    PthreadAttr* {.byref, importc: "pthread_attr_t", header: "<sys/types.h>".} = object
-    PthreadBarrier* {.byref, importc: "pthread_barrier_t", header: "<sys/types.h>".} = object
+    PthreadAttr* {.importc: "pthread_attr_t", header: "<sys/types.h>".} = object
+      when (defined(linux) and not defined(android)) and defined(amd64):
+        abi: array[56 div sizeof(clong), clong]
```

Comment:
I would create `consts` to hold the size and add checks at global initialization similar to: https://github.com/mratsim/photon-jit/blob/747ae2db3698e1b814e78fded175d08a778fada9/photon_jit/photon_osalloc.nim#L21-L77=


```Nim
when defined(windows):
  type
    MemAlloc* {.size: cint.sizeof.} = enum
      # is a flag on Win32!
      MemCommit = 0x1000 # Allocate immediately
    MemFree* {.size: cint.sizeof.} = enum
      # is *not* a flag on Win32!
      MemRelease = 0x1000 # Free immediately
elif defined(osx) or defined(ios) or defined(bsd):
  # Note: MacOS and Iphone uses MAP_ANON instead of MAP_ANONYMOUS
  # They also define MAP_JIT= 0x0800
  type MemMap* {.size: cint.sizeof.} = enum
    MapPrivate = 0x02       # Changes are private
    MapAnonymous = 0x1000   # Don't use a file
elif defined(solaris):
  type MemMap* {.size: cint.sizeof.} = enum
    MapPrivate = 0x02
    MapAnonymous = 0x100
elif defined(haiku):
  type MemMap* {.size: cint.sizeof.} = enum
    MapPrivate = 0x02
    MapAnonymous = 0x08
else: # ASM-Generic
  # Note, Nim splits linux x86-64 and the rest
  # This is at least valid on Android ARM
  # unsure about MIPS and co
  type MemMap* {.size: cint.sizeof.} = enum
    MapPrivate = 0x02
    MapAnonymous = 0x20

when not defined(release):
  when defined(windows):
      var DebugMemCommit {.importc: "MEM_COMMIT", header: "<windows.h>".}: cint
      var DebugMemRelease {.importc: "MEM_RELEASE", header: "<windows.h>".}: cint

      assert ord(MemCommit) == DebugMemCommit, "Your CPU+OS platform is misconfigured"
      assert ord(MemRelease) == DebugMemRelease, "Your CPU+OS platform is misconfigured"
  else:
    block:
      var DebugMapPrivate {.importc: "MAP_PRIVATE", header: "<sys/mman.h>".}: cint
      var DebugMapAnonymous {.importc: "MAP_ANONYMOUS", header: "<sys/mman.h>".}: cint

      assert ord(MapPrivate) == DebugMapPrivate, "Your CPU+OS platform is misconfigured"
      assert ord(MapAnonymous) == DebugMapAnonymous, "Your CPU+OS platform is misconfigured"
```

Reply:
yeah, this is based on `posix.nim` - but indeed, constants would probably make sense, also because the actual header file uses constants

Reply:
I thought a bit more about the consts, but that doesn't work in this case: the ABI requires a specific alignment as well as size, meaning that the type used for the abi is significant (in the header file, this is solved with a union)